### PR TITLE
[SPARK-24198][SparkR][SQL] Adding slice function to SparkR

### PR DESCRIPTION
**Pull request by mn-mikke**
## What changes were proposed in this pull request?
The PR adds the `slice` function to SparkR. The function returns a subset of consecutive elements from the given array.
```
> df <- createDataFrame(cbind(model = rownames(mtcars), mtcars))
> tmp <- mutate(df, v1 = create_array(df$mpg, df$cyl, df$hp))
> head(select(tmp, slice(tmp$v1, 2L, 2L)))
```
```
  slice(v1, 2, 2)                                                               
1          6, 110
2          6, 110
3           4, 93
4          6, 110
5          8, 175
6          6, 105
```

## How was this patch tested?

A test added into R/pkg/tests/fulltests/test_sparkSQL.R

### CODE REVIEW

**Comment by mn-mikke** [comment]
cc @HyukjinKwon @felixcheung

**Comment by HyukjinKwon** [comment]
@mn-mikke, let's do this with few more functions together next time per the suggestions from @felixcheung and @shivaram who're PMCs used to R side better than me.

File: R/pkg/R/functions.R [hunk 1]
```diff
@@ -3124,6 +3125,23 @@ setMethod("size",
...
+#' (or starting from the end if start is negative) with the specified length.
+#'
+#' @rdname column_collection_functions
+#' @param start an index indicating the first element occuring in the result.
+#' @param length a number of consecutive elements choosen to the result.
+#'
+#' @aliases slice slice,Column-method
+#' @note slice since 2.4.0
+setMethod("slice",
+signature(x = "Column"),
```
Comment (felixcheung) [inline R/pkg/R/functions.R]: please fix the ident - see "size" for example

**Review by felixcheung — COMMENTED** [review]
yes, and let's add a bunch at a time

**Comment by mn-mikke** [comment]
Oh, sorry for that. I'll group several functions next time. Thanks guys.

File: R/pkg/R/functions.R [hunk 2]
```diff
@@ -3138,6 +3139,23 @@ setMethod("size",
...
           })
 
+#' @details
+#' \code{slice}: Returns an array containing all the elements in x from the index start
+#' (or starting from the end if start is negative) with the specified length.
+#'
+#' @rdname column_collection_functions
+#' @param start an index indicating the first element occuring in the result.
+#' @param length a number of consecutive elements choosen to the result.
+#'
```
Comment (felixcheung) [inline R/pkg/R/functions.R]: nit: remove unneeded line

**Review by felixcheung — APPROVED** [review]
LGTM

**Comment by HyukjinKwon** [comment]
lgtm too except that the nit.

**Pull request closed without merging**